pkg/reconciler/instances/warden/volumemounts.go

```diff
@@ -0,0 +1,121 @@
+package warden
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/kubernetes"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const wardenAdmissionDeploymentName = "warden-admission"
+const wardenAdmissionDeploymentNamespace = "kyma-system"
+const volumeName = "certs"
+
+type CleanupWardenAdmissionCertColumeMounts struct {
+	name string
+}
+
+func (a *CleanupWardenAdmissionCertColumeMounts) Run(context *service.ActionContext) error {
+	k8sClient := context.KubeClient
+
+	targetImage := getWardenAdmissionTargetImage(context.Task.Configuration)
+
+	if isQualifiedForCleanup(targetImage) {
+
+		deployment, err := getDeployment(context.Context, k8sClient, wardenAdmissionDeploymentName, wardenAdmissionDeploymentNamespace)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("while checking if %s deployment is present on cluster", wardenAdmissionDeploymentName))
+		}
+		if deployment != nil {
+
+			volumeIndex := getVolumeIndexByName(deployment, volumeName)
+			volumeMountIndex := getVolumeMountIndexByName(deployment, volumeName)
+
+			if volumeIndex == -1 || volumeMountIndex == -1 {
+				context.Logger.Debugf("no action needed for warden admission deployment before applying image %s", targetImage)
+				return nil
+			}
+
+			context.Logger.Debugf("warden admission deployment qualifies for Volume[%d] nad VolumeMount[%d] cleanup before applying image %s", volumeIndex, volumeMountIndex, targetImage)
+			data := fmt.Sprintf(`[{"op": "remove", "path": "/spec/template/spec/containers/0/volumeMounts/%d"},{"op": "remove", "path": "/spec/template/spec/volumes/%d"}]`, volumeMountIndex, volumeIndex)
```

Comment:
It would be nice to have an unit-tested function that creates this JSON-patch data.

Reply:
What do you mean?
There is a mock call verification in the unit test (if expected call is recorded)
Wouldn't testing the line 47 be just testing if `fmt.Sprintf` works? I'd like to avoid testing that tbh.